addOns/retest/src/main/java/org/zaproxy/addon/retest/RetestPlanGenerator.java

```diff
@@ -0,0 +1,141 @@
+
+public class RetestPlanGenerator {
+
+    List<AlertData> alertDataList;
+
+    public RetestPlanGenerator(List<AlertData> alertDataList) {
+        this.alertDataList = alertDataList;
+    }
+
+    public AutomationPlan getPlan() {
+        AutomationProgress planProgress = new AutomationProgress();
+        AutomationEnvironment planEnv = getEnvironment(planProgress);
+        List<AutomationJob> planJobs = getAllJobs(planEnv);
+        return new AutomationPlan(planEnv, planJobs, planProgress);
+    }
+
+    private AutomationEnvironment getEnvironment(AutomationProgress progress) {
+        AutomationEnvironment env = new AutomationEnvironment(progress);
+        AutomationEnvironment.Data data = env.getData();
+
+        ContextWrapper.Data cwd = new ContextWrapper.Data();
+        cwd.setName("Retest Plan");
+        cwd.setUrls(alertDataList.stream().map(AlertData::getUrl).collect(Collectors.toList()));
+        List<ContextWrapper.Data> contextList = new ArrayList<>();
+        contextList.add(cwd);
+        data.setContexts(contextList);
+
+        env.addContext(cwd);
+        return env;
+    }
+
+    private List<AutomationJob> getAllJobs(AutomationEnvironment env) {
+        List<AutomationJob> jobs = new ArrayList<>();
+        jobs.add(new PassiveScanConfigJob());
+        jobs.add(getRequestorJob());
+        alertDataList.stream()
+                .filter(t -> t.getAlert().getSource().equals(Alert.Source.ACTIVE))
+                .map(this::getActiveScanJob)
+                .forEach(jobs::add);
+        jobs.add(getPassiveScanWaitJob());
+        jobs.forEach(t -> t.setEnv(env));
+        return jobs;
+    }
+
+    private RequestorJob getRequestorJob() {
+        RequestorJob reqJob = new RequestorJob();
+        List<RequestorJob.Request> requests =
+                alertDataList.stream()
+                        .map(
+                                t ->
+                                        new RequestorJob.Request(
+                                                t.getUrl(),
+                                                t.getAlertName(),
+                                                t.getMethod(),
+                                                t.getMsg().getRequestBody().toString(),
+                                                null))
+                        .collect(Collectors.toList());
+        reqJob.getData().setRequests(requests);
+        return reqJob;
+    }
+
+    private ActiveScanJob getActiveScanJob(AlertData alertData) {
+        ActiveScanJob activeScanJob = new ActiveScanJob();
+        activeScanJob.getParameters().setContext("Retest Plan");
+        activeScanJob.getData().getPolicyDefinition().setDefaultThreshold("Off");
+        List<ActiveScanJob.Rule> rules = new ArrayList<>();
+        rules.add(
+                new ActiveScanJob.Rule(
+                        alertData.getScanRuleId(), alertData.getAlertName(), "Medium", "Medium"));
```

Comment:
This should be documented in the help, that the current assumption is Medium/Medium.

Ideally this should be user selectable (unless it actually is recorded on the original alert [I don't think it is]).